packages/fx-core/src/plugins/resource/bot/index.ts

```diff
@@ -145,6 +145,45 @@ export class TeamsBot implements Plugin {
     );
   }
 
+  private wrapError(
+    e: any,
+    context: PluginContext,
+    sendTelemetry: boolean,
+    name: string
+  ): FxResult {
+    if (e.innerError) {
+      e.message += ` Detailed error: ${e.innerError.message}.`;
+      if (e.innerError.response?.data?.errorMessage) {
+        e.message += ` Reason: ${e.innerError.response?.data?.errorMessage}`;
+      }
+    }
+    Logger.error(e.message);
+    if (e instanceof UserError || e instanceof SystemError) {
+      const res = err(e);
+      sendTelemetry && telemetryHelper.sendResultEvent(context, name, res);
+      return res;
+    }
+
+    if (e instanceof PluginError) {
+      const result =
+        e.errorType === ErrorType.System
+          ? ResultFactory.SystemError(e.name, e.genMessage(), e.innerError)
+          : ResultFactory.UserError(e.name, e.genMessage(), e.showHelpLink, e.innerError);
+      sendTelemetry && telemetryHelper.sendResultEvent(context, name, result);
+      return result;
+    } else {
+      // Unrecognized Exception.
+      const UnhandledErrorCode = "UnhandledError";
+      sendTelemetry &&
+        telemetryHelper.sendResultEvent(
+          context,
+          name,
+          ResultFactory.SystemError(UnhandledErrorCode, `Got an unhandled error: ${e.message}`)
```

Comment:
It's not recommend to send telemetry with unhandled error messages. We can't ensure the message doesn't contain user information.